File: demo-samples/sample.js
// Adds two numbers together
function add(a, b) {
  return a + b;
}

/**
 * Multiplies two numbers.
 * This block comment spans multiple lines.
 */
function multiply(a, b) {
  return a * b;
}

// Edge case: these look like comment markers but are just string values
const commentMarkers = ['/*', '*/', '//'];

// don't let apostrophes inside a comment break the scanner
function greet(name) {
  return `Hello, ${name}!`; // trailing comment after a template literal
}

module.exports = { add, multiply, greet };

File: demo-samples/sample.py
# Returns the factorial of n
def factorial(n):
    if n <= 1:
        return 1
    return n * factorial(n - 1)


"""
This is a module-level docstring style comment
used to test multi-line block detection.
"""


def greet(name):
    # don't forget: apostrophes shouldn't break parsing
    message = "hello, it's a great day"  # trailing comment
    return f"{message}, {name}!"


# Edge case: a string that looks like a comment marker
fake_comment = "# this is not a comment, just a string"
